lib/rubocop/cop/rspec/capybara/specific_actions.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for there is a more specific actions offered by Capybara.
+        #
+        # @example
+        #
+        #   # bad
+        #   find('a').click
+        #   find('button.cls').click
+        #   find('a', exact_text: 'foo').click
+        #   find('div button').click
+        #
+        #   # good
+        #   click_link
+        #   click_button(class: 'cls')
+        #   click_link(exact_text: 'foo')
+        #   find('div').click_button
+        #
+        class SpecificActions < Base
+          MSG = "Prefer `%<good_action>s` over `find('%<selector>s').click`."
+          RESTRICT_ON_SEND = %i[click].freeze
+          SPECIFIC_ACTION = {
+            'button' => 'button',
+            'a' => 'link'
+          }.freeze
+
+          # @!method first_argument(node)
+          def_node_matcher :first_argument, <<-PATTERN
+            (send _ :find (str $_) ...)
+          PATTERN
+
+          # @!method option?(node)
+          def_node_search :option?, <<-PATTERN
+            (pair (sym %) _)
+          PATTERN
+
+          def on_send(node)
+            return unless (receiver = receiver(node))
+
+            first_argument(receiver) do |arg|
+              return unless supported_selector?(arg)
+              return unless (selector = last_selector(arg)[/^\w+/, 0])
+              return unless (action = specific_action(selector))
+              next unless specific_action_option?(receiver, arg, action)
+              next unless specific_action_pseudo_classes?(arg)
+
+              range = offense_range(node, receiver)
+              add_offense(range, message: message(action, selector))
+            end
+          end
+
+          def receiver(node)
+            return unless (receiver = node.receiver)
+
+            receiver if receiver.method?(:find)
```

Comment:
the method name is later checked in the first_argument matcher as well. I'd suggest using just the matcher as following: